spec/request/admin/pages_spec.rb

```diff
@@ -155,5 +155,23 @@
       expect( response.body ).to     include 'updated-by-test'
       expect( response.body ).not_to include old_slug
     end
+
+    it 'shows the appropriate input type for each element type' do
+      allow( PageElement ).to receive_message_chain(
+        :select_filenames
+      ).and_return( %w[ FILE.png image.jpeg image.gif ] )
+
+      page = create :page_with_one_of_each_element_type
+
+      get admin_page_path( page )
+
+      expect( response      ).to have_http_status :ok
+      expect( response.body ).to include I18n.t( 'admin.pages.edit_page' ).titlecase
+      expect( response.body ).to match %r{<input [^>]*value="SHORT!"[^>]*>}
+      expect( response.body ).to match %r{<textarea [^>]+>\nLONG!</textarea>}
+      expect( response.body ).to match %r{<option [^>]+>FILE.png</option>}
+      # TODO: improve this test once the WYSIWYG editor code is in place
+      expect( response.body ).to match %r{<textarea [^>]+>\nHTML!</textarea>}
```

Comment:
These expectations look like a good candidate for simplifying with Capybara's `have_css` and `have_field` matchers.

```diff
-expect( response.body ).to match %r{<input [^>]*value="SHORT!"[^>]*>}
+expect( response.body ).to have_field 'the-field-name', with: 'SHORT!'
```

To make these matchers available to request specs needs the following config:

```rb
RSpec.configure do |config|
  # Make have_* matchers available to request specs e.g.:
  # expect(response.body).to have_title 'My Page Title'
  config.include Capybara::RSpecMatchers, type: :request

  # Or sometimes its useful to make these matchers available to other spec types:
  # config.include Capybara::RSpecMatchers, type: /mailer|model|request/
end
```